pages/api/module/patch.ts

```diff
@@ -0,0 +1,25 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { Module } from "../models/module";
+
+const patch = async (
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> => {
+    const { id } = req.query;
+    const module = await Module.findById(id).exec();
+
+    if (!module)
+        return res
+            .status(404)
+            .send("The module with the given ID was not found.");
+
+    for (const key in req.body) {
```

Comment:
I've seen this update function used in almost all endpoint pr's it would be good to have util file and keep functions like these there so we don't have to keep writing the same code. Not a huge deal if it takes too long though

Reply:
let's add a TODO for this (comment in the codebase)

Reply:
let's add a TODO for this (comment in the codebase)